src/stores/problem_sets.ts

```diff
@@ -25,15 +26,33 @@ export const useProblemSetStore = defineStore('problem_sets', {
 		async fetchProblemSets(course_id: number): Promise<void> {
 			const response = await api.get(`courses/${course_id}/sets`);
 			const sets_to_parse = response.data as Array<ParseableProblemSet>;
+			logger.debug(`[problem_sets/fetchProblemSets] parsing response: ${sets_to_parse.join(', ')}`);
 			this.problem_sets = sets_to_parse.map((set) => parseProblemSet(set));
 		},
-		async updateSet(set: ProblemSet): Promise<void> {
+		async updateSet(set: ProblemSet): Promise<{ error: boolean; message: string }> {
 			const sessionStore = useSessionStore();
 			const course_id = sessionStore.course.course_id;
 			// shouldn't we be throwing an error if set_id is null or 0?
 			const response = await api.put(`courses/${course_id}/sets/${set.set_id ?? 0}`, set.toObject());
-			const updated_set = parseProblemSet(response.data as ParseableProblemSet);
-			const index = this.problem_sets.findIndex(s => s.set_id === set.set_id);
+			let message = '';
+			let is_error = false;
+			if (response.status === 200) {
+				const updated_set = parseProblemSet(response.data as ParseableProblemSet);
+				if (JSON.stringify(updated_set) !== JSON.stringify(set)) {
+					logger.error('[updateSet] response does not match requested update to set');
```

Comment:
should we pass this error back to the component where it is called?

Reply:
If we get this error, there are much bigger problems...

To get to this point, our object must have passed validation to be submitted in the first place and then the backend must have successfully completed our requested update. If there's any difference here, we should log the error to the backend (`logger.error`) and update the data to match what the DB responded with.

I guess what I'm saying is that if this error occurs, we have things to fix before release. It's not something to worry about bubbling up to the UI for the user. There's nothing they can do about it.